ocp_utilities/must_gather.py

```diff
@@ -51,3 +54,41 @@ def run_must_gather(
         flag_string = "".join([f" --{flag_name}" for flag_name in flag_names])
         base_command += f" {flag_string}"
     return run_command(command=shlex.split(base_command), check=False)[1]
+
+
+def collect_must_gather(must_gather_output_dir, kubeconfig_path, cluster_name, product_name):
+    """
+    Run must-gather command with an option to create target directory using must_gather_output_dir, cluster_name, product_name.
+    (This function can be used for running must-gather incase any product installation fails.)
+
+    Args:
+        must_gather_output_dir (str): Path to base directory where must-gather logs will be stored
+        kubeconfig_path (str): Path to kubeconfig
+        cluster_name (str): Cluster Name for which must-gather will run
+        product_name (str): Product Name installed on given cluster
+    """
+
+    target_dir = os.path.join(must_gather_output_dir, "must-gather", product_name, cluster_name)
+
+    try:
+        if not os.path.exists(kubeconfig_path):
```

Comment:
`run_must_gather` do not force 'kubeconfig', why you force it here?

Reply:
Not necessarily, if you have export `KUBECONFIG` it will work.

Reply:
Updated the code to use `os.environ.get('KUBECONFIG')` incase `kubeconfig_path` is None